### ISSUE TITLE
The Google login button is partly functional although the feature is disabled

### ISSUE THREAD

**Issue by Tonev** [brave/brave-browser#17540]
## Description 

I noticed that I could use the `Sign up with Google` button on websites although the `Allow Google login buttons on third party sites` is disabled at `brave://settings/socialBlocking`.

It's quite important to note that users **can't** successfully sign up when the feature is disabled, but regardless of that it's wrong that the button is not grayed out.

## Steps to Reproduce

   1. Disable `Allow Google login buttons on third party sites` at `brave://settings/socialBlocking`.
   2. Go on Twitter or another website where you are able to sign up with Google.
   3. Click on the Google button.

## Actual result:

https://user-images.githubusercontent.com/20608253/129837528-8d96cd7f-266a-45a4-a9d1-72a26f613781.mp4

## Expected result:
The button shouldn't be partly functional, it should be grayed out (unavailable) from the beginning as long as the feature is disabled.

## Reproduces how often: 

Easily

## Brave version (brave://version info)

<html>
<body>

Brave | 1.28.105 Chromium: 92.0.4515.131 (Official Build) (64-bit)
-- | --
Revision | 6b8d6c56ce21e38a72f7c4becb5abc1fa5134f29-refs/branch-heads/4515@{#1933}
OS | Windows 7 Service Pack 1 (Build 7601)
JavaScript | V8 9.2.230.22
User Agent | Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36
Command Line | "C:\Program Files\BraveSoftware\Brave-Browser\Application\brave.exe" --enable-dom-distiller --disable-domain-reliability --no-pings --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --lso-url=https://no-thanks.invalid --variations-server-url=https://variations.brave.com/seed --enable-features=LegacyTLSEnforced,AutoupgradeMixedContent,WinrtGeolocationImplementation,SafetyTip,WebUIDarkMode,PrefetchPrivacyChanges,PasswordImport,ReducedReferrerGranularity --disable-features=AutofillServerCommunication,FirstPartySets,FlocIdComputedEventLogging,HandwritingRecognitionWebPlatformApiFinch,TrustTokens,DirectSockets,EnableProfilePickerOnStartup,TextFragmentAnchor,AutofillEnableAccountWalletStorage,FledgeInterestGroupAPI,IdleDetection,InterestCohortFeaturePolicy,FledgeInterestGroups,NotificationTriggers,WebOTP,LiveCaption,PrivacySandboxSettings,FederatedLearningOfCohorts,SignedExchangeSubresourcePrefetch,NetworkTimeServiceQuerying,SubresourceWebBundles,SignedExchangePrefetchCacheForNavigations,InterestCohortAPIOriginTrial,HandwritingRecognitionWebPlatformApi,LangClientHintHeader,EnablePasswordsAccountStorage --flag-switches-begin --enable-features=LegacyTLSEnforced,AutoupgradeMixedContent,WinrtGeolocationImplementation,SafetyTip,WebUIDarkMode,PrefetchPrivacyChanges,PasswordImport,ReducedReferrerGranularity,EphemeralStorage --flag-switches-end
Executable Path | C:\Program Files\BraveSoftware\Brave-Browser\Application\brave.exe
Profile Path | C:\Users\User\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default
Variations | AdRewardsStudy:NextPaymentDayEphemeralStorageStudy:EnabledNativeCosmeticFilteringStudy:EnabledPermissionLifetimeReleaseStudy:EnabledSpeedreaderReleaseStudy:Disabled

</body>
</html>

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Does the issue resolve itself when disabling Brave Rewards? N/A
- Is the issue reproducible on the latest version of Chrome? N/A

**Comment by Tonev** [comment]
I did more testing on Android and it looks like the issue is present there also. I tested `Brave 1.28.105`, my phone is `Huawei Mate 10 lite`, running on Android 8.

The `Sign up with Google` button in the mobile version of Twitter remained unavailable regardless of the state of `Allow Google login buttons on third party sites`, but I encountered the same behavior as on desktop (Windows) on other websites, so I believe the same issue is present on Android as well. I will appreciate if someone could add the missing `OS/Android` label whenever they can.
